packages/autocomplete-plugin-recent-searches/src/createLocalStorageRecentSearchesPlugin.ts

```diff
@@ -0,0 +1,65 @@
+import { AutocompletePlugin } from '@algolia/autocomplete-core';
+
+import {
+  createRecentSearchesPlugin,
+  CreateRecentSearchesPluginParams,
+  RecentSearchesPluginData,
+} from './createRecentSearchesPlugin';
+import { RecentSearchesItem } from './types';
+import {
+  LOCAL_STORAGE_KEY,
+  createLocalStorage,
+  search as defaultSearch,
+  SearchParams,
+} from './usecases/localStorage';
+
+export type CreateRecentSearchesLocalStorageOptions<
+  TItem extends RecentSearchesItem
+> = {
+  /**
+   * The unique key to name the store of recent searches.
+   *
+   * @example "top_searchbar"
+   */
+  key: string;
+
+  /**
+   * The number of recent searches to store.
+   *
+   * @default 5
+   */
+  limit?: number;
```

Comment:
the other api is using hitsPerPage, but this doesn't have pagination per se. Could it make sense to have `hitsPerPage` already even if pagination isn't yet implemented?

Reply:
Can you elaborate?

Reply:
I saw a bit lower in this file hitsPerPage being around searchParameters, but here we are using limit for the argument. If we call this `hitsPerPage`, the API is one word smaller I think

Reply:
`hitsPerPage` is a concept in Algolia, and the Recent Searches plugin is not tied to Algolia, so I'm not sure about that.

This makes me think that the `data` properties that we have should probably contain the word `Algolia`. I'll change that.